utils/evaluateResponse.ts

```diff
@@ -0,0 +1,123 @@
+import { expect } from "@jest/globals";
+import baseTypePatterns from "./baseTypePatterns";
+import fixtures from "../fixtures";
+
+// A robust version of the "typeof" operator
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof#custom_method_that_gets_a_more_specific_type
+function type(value) {
+  if (value === null) {
+    return "null";
+  }
+
+  const baseType = typeof value;
+  // Primitive types
+  if (!["object", "function"].includes(baseType)) {
+    return baseType;
+  }
+
+  // Symbol.toStringTag often specifies the "display name" of the
+  // object's class. It's used in Object.prototype.toString().
+  const tag = value[Symbol.toStringTag];
+  if (typeof tag === "string") {
+    return tag;
+  }
+
+  // If it's a function whose source code starts with the "class" keyword
+  if (
+    baseType === "function" &&
+    Function.prototype.toString.call(value).startsWith("class")
+  ) {
+    return "class";
+  }
+
+  // The name of the constructor; for example `Array`, `GeneratorFunction`,
+  // `Number`, `String`, `Boolean` or `MyCustomClass`
+  const className = value.constructor.name;
+  if (typeof className === "string" && className !== "") {
+    return className;
+  }
+
+  // At this point there's no robust way to get the type of value,
+  // so we use the base implementation.
+  return baseType;
+}
+
+function testString(value) {
+  const patternFound = baseTypePatterns
+    .find(baseTypePattern => 
+      value
+        .match(baseTypePattern)
+    );
```

Comment:
Correct me if I'm wrong - don't we want to test each string against a specific regex pattern? 
Is it sufficient that the string matches any of the patterns?

Reply:
maybe create a dictionary of the response's keys with its corresponding type and use that to determine what each field expects as far as pattern type? @yurikoinaba

Reply:
Yeah, perhaps something like
```
evaluateResponse(await eth_blockNumber(), {
   jsonrpc: "2.0",
   id: 1,
   result: baseTypes.uint.pattern
});
  ```
though that is similar to the pattern you had before.

Curious to hear what Shaun had in mind.

Reply:
I think passing the pattern would work like this for simple case but as we go into a more complicated ones like so:

**eth_getBlockByNumber**
```
{
    "jsonrpc": "2.0",
    "id": 1,
    "result": {
        "number": "0x8286",
        "hash": "0xef4c45f27c708c46daf9887150d8ee4f771ef79f863e252e450842cb365a0071",
        "parentHash": "0x75913fc9590e20890f4bade40c0bf3336f41075c092c4a0d66353efd9e581c96",
        "logsBloom": "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
        "transactionsRoot": "0xb53557c748eda383362367dfe1f1ea8f00afbbf849ecad3c030cbaaf06484ce5",
        "stateRoot": "0x28fdd808e04775d98b7ec9a970d122b81a17a6c16ce99244f7f3a731f3ebd49d",
        "receiptsRoot": "0x139b070d62a2271c359a59eebab592ebe2514727593f696c84e81b9012d15a77",
        "miner": "0x144e0fe5e69893577107a15a7c76babd59f0a279",
        "size": "0x1fcf5",
        "gasLimit": "0x1c9c380",
        "gasUsed": "0x16ee5c8",
        "timestamp": "0x643fee5e",
        "transactions": [
            {
                "blockHash": "0xef4c45f27c708c46daf9887150d8ee4f771ef79f863e252e450842cb365a0071",
                "blockNumber": "0x8286",
                "transactionIndex": "0x0",
                "hash": "0xcf8161141a28f3ebd3abbbfa34bc5f8bd1c2c90a9443b66436eca8b1eb6c39b8",
                "type": "0x2",
                "nonce": "0x40",
                "from": "0xe33693548d5849845c9915342ac8ad932a4b2f36",
                "to": "0x3d5f994c679244b3230d8c884f67752ff94e7aa5",
                "value": "0x64",
                "input": "0x",
                "gas": "0x38270",
                "gasPrice": "0x35a4e900",
                "maxPriorityFeePerGas": "0x35a4e900",
                "maxFeePerGas": "0x35a4e900",
                "chainId": "0x67f",
                "v": "0x1c",
                "r": "0xbae0258ea212ab6411be224a8f1b3e169c9b3cd04a2b9bcabe0488a3f640a2fa",
                "s": "0x7a5d321e958ce27f20cedfb09f04070695f6b6b51fee5f9b4746b8fc5fc5715f",
                "accessList": []
            },
            ...
        ],
}
```

the complexity would quickly increase as what we've seen in what I did previously. but I think yes if we do create a dictionary of *key* => *pattern* like:

```
"parentHash": "bytes32",
"number": "uint",
"sha3Uncles": "hash32",
```

I think the library should be able to perform a look up and figure out if the value passed for each key contains the correct schema type or not therefore pass/fail it. This change should be straightforward as I already have the base code that performs the evaluation. whew! 🙂

Reply:
Hey @rabyyuson , with your last comment i think that describes what I had in mind, i.e each test goes through the result verifying all the keys are there and all the values match the regex pattern according to the definition in those yaml schema files. I think you shouldn't have to create such a dictionary because those yaml schema files are basically already that very thing. Or at the very least you can create dictionary from those yaml schema files and you should be able to do it programmatically, not by hand. In this case, it would be one dictionary per rpc method, a dictionary for eth_getBlock, another one for eth_getTransactionReciept, etc.